### PR TITLE
[X86][MC] Not emit {evex} for VEX-promoted instructions with GPR operands

### PR DESCRIPTION
**Pull request by KanRobert**
To align with
1. GNU binutils's behavior for APX instructions
2. LLVM's behaviour for EVEX intructions with VEX variant

### CODE REVIEW

File: llvm/test/MC/X86/apx/kmov-att.s [hunk 1]
```diff
@@ -3,6 +3,21 @@
...
+         {evex}	kmovb	%k1, %k2
+# CHECK: {evex}	kmovw	%k1, %k2
+# CHECK: encoding: [0x62,0xf1,0x7c,0x08,0x90,0xd1]
+         {evex}	kmovw	%k1, %k2
+# CHECK: {evex}	kmovd	%k1, %k2
+# CHECK: encoding: [0x62,0xf1,0xfd,0x08,0x90,0xd1]
+         {evex}	kmovd	%k1, %k2
+# CHECK: {evex}	kmovq	%k1, %k2
+
+# CHECK-NOT: {evex}
```
Comment (XinWang10) [inline llvm/test/MC/X86/apx/kmov-att.s:17]: Does this check in the wrong line, it maybe in line 21 I think.

Comment (KanRobert) [reply llvm/test/MC/X86/apx/kmov-att.s:17]: You're right!

Comment (KanRobert) [reply llvm/test/MC/X86/apx/kmov-att.s:19]: Done.

**Review by XinWang10 — APPROVED** [review]
LGTM

**Pull request merged**